devito/builtins.py

```diff
@@ -364,6 +367,29 @@ def buff(i, j):
 
     assign(lhs, rhs, options=options, name=name, **kwargs)
 
+    if pad_halo:
+        pad_fd_halo(function)
+
+
+def pad_fd_halo(function):
+    """ Pad finite-difference halo with edge values."""
+    h_shape = function._data_with_outhalo.shape
+    for i, h in enumerate(function._size_outhalo):
+        if h.left == 0 and h.right == 0:
+            continue
+        slices = [slice(None)]*len(function.shape)
+        slices_d = [slice(None)]*len(function.shape)
+        # left part
+        slices[i] = slice(0, h.left)
```

Comment:
this works ok with mpi where on a rank you could end up with
```
        slices[i] = slice(0, 0)
        slices_d[i] = slice(0, 1, 1)
```
?

Reply:
Same for the right bit.

Reply:
Not sure I follow

Reply:
Fixed MPI case